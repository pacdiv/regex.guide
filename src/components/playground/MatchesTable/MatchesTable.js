import PropTypes from 'prop-types'
import React from "react"

import { ErrorText, Row, TableContainer } from "./MatchesTable.style"

function getMatchesArray(sourceString, regexChunks, flags) {
  const regex = new RegExp(regexChunks.join(""), flags.global ? "g" : "")
  return sourceString.match(regex)
}

function MatchesTable({ flags, regexChunks, sourceString }) {
  const matches = getMatchesArray(sourceString, regexChunks, flags)

  if (!matches) {
    return <ErrorText>The regex gets no matches.</ErrorText>
  }

  return (
    <TableContainer>
      <p>Matches:</p>
      <div className="table">
        <Row>
          <div>#</div>
          <div>content</div>
        </Row>
        {matches.map((match, index) => (
          <Row key={`regex-match-${index}`}>
            <div>{index}</div>
            <div>{match}</div>
          </Row>
        ))}
      </div>
    </TableContainer>
  )
}

MatchesTable.propTypes = {
  flags: PropTypes.shape({
    global: PropTypes.bool
  }),
  regexChunks: PropTypes.arrayOf(PropTypes.string),
  sourceString: PropTypes.string
}

MatchesTable.defaultProps = {
  flags: {},
  regexChunks: [],
  sourceString: ""
}

export default MatchesTable
