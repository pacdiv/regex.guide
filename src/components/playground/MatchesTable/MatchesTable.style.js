import styled from "@emotion/styled"

export const ErrorText = styled.p`
  color: crimson;
  margin: 0.5em 0;
`

export const TableContainer = styled.div`
  margin: 1em 0 0;

  p {
    margin-bottom: 0.5em;
  }

  div.table {
    align-items: center;
    border: 1px solid #f0f0f0;
    border-radius: 4px;
    display: flex;
    flex-direction: column;
    justify-content: stretch;
    margin: 0 auto 2em;
    max-width: 16em;
    width: 100%;
  }
`

export const Row = styled.div`
  align-items: center;
  display: flex;
  flex-direction: row;
  width: 100%;

  &:not(:last-of-type) {
    border-bottom: 1px solid #f0f0f0;
  }

  > div {
    &:first-of-type {
      min-width: 3em;
    }
    &:last-of-type {
      border-left: 1px solid #f0f0f0;
      flex: 1 1 auto;
    }
  }
`
