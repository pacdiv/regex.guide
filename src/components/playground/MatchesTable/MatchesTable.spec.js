import React from "react"
import { render } from "@testing-library/react"
import "@testing-library/jest-dom/extend-expect"

import MatchesTable from "./MatchesTable"

test("MatchesTable component with results", async () => {
  const { container } = render(
    <MatchesTable
      flags={{ global: true }}
      regexChunks={["\\d+"]}
      sourceString="If you donâ€™t succeed at first, call it v1.0 ðŸ¤«"
    />
  )

  expect(container).toMatchSnapshot()
})

test("MatchesTable component without results", async () => {
  const { container } = render(
    <MatchesTable
      flags={{ global: true }}
      regexChunks={["\\d+"]}
      sourceString="â€¦ I'm kidding!"
    />
  )

  expect(container).toMatchSnapshot()
})
