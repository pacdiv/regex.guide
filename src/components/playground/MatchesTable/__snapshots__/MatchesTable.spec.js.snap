// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MatchesTable component with results 1`] = `
<div>
  <div
    class="css-d550xy"
  >
    <p>
      Matches:
    </p>
    <div
      class="table"
    >
      <div
        class="css-140sd30"
      >
        <div>
          #
        </div>
        <div>
          content
        </div>
      </div>
      <div
        class="css-140sd30"
      >
        <div>
          0
        </div>
        <div>
          1
        </div>
      </div>
      <div
        class="css-140sd30"
      >
        <div>
          1
        </div>
        <div>
          0
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`MatchesTable component without results 1`] = `
<div>
  <p
    class="css-7jl3jq"
  >
    The regex gets no matches.
  </p>
</div>
`;
